Make service registration (optionally) interactive

palango opened the pull request:
First steps towards #758 
Mainly opening this to start a discussion.

This makes the registration script optionally interactive for use in the RSB.

This looks currently like:
```
--------------------------------------------------------------------------
| This is an Alpha version of experimental open source software          |
| released as a test version under an MIT license and may contain        |
| errors and/or bugs. No guarantee or representations whatsoever is      |
| made regarding its suitability (or its use) for any purpose or         |
| regarding its compliance with any applicable laws and regulations.     |
| Use of the software is at your own risk and discretion and by          |
| using the software you acknowledge that you have read this             |
| disclaimer, understand its contents, assume all risk related           |
| thereto and hereby release, waive, discharge and covenant not to       |
| sue Brainbot Labs Establishment or any officers, employees or          |
| affiliates from and for any direct or indirect liability resulting     |
| from the use of the software as permissible by applicable laws and     |
| regulations.                                                           |
|                                                                        |
| This script will help you to register at the Raiden Service Registry,  |
| see https://raiden-network-specification.readthedocs.io/en/latest/     |
| Service Contracts for more information.                                |
| The Raiden Service Registry is designed to hold RDN tokens as          |
| deposit. At the end of the duration of a slot (6 months) you will get  |
| your tokens back to the same address used for bidding for that slot.   |
| DO NOT LOSE YOUR PRIVATE KEY.                                          |
--------------------------------------------------------------------------


This will run the registration with the address 0xBF1dABcf9eCA2D89fB24be3E6E45Ef422eC39ca0
	See https://goerli.etherscan.io/address/0xBF1dABcf9eCA2D89fB24be3E6E45Ef422eC39ca0

The address of the service registry contract used is 0x735722704E365c37247Bb3e4eC52D6c937C54539
	See https://goerli.etherscan.io/address/0x735722704E365c37247Bb3e4eC52D6c937C54539

The address of the token used is 0xc116edAD88cda44E703ef1fc59766268E4aa187B
	See https://goerli.etherscan.io/address/0xc116edAD88cda44E703ef1fc59766268E4aa187B
The account balance of that token is 0

The current required deposit is 191
	Note: The required deposit may change over time.

Sending transaction: Minting new Test RDN tokens
	See https://goerli.etherscan.io/tx/0x07e39095db40b8f677652e2664a5701ccd9a56ae78684c036c32645d64a956be

Sending transaction: Allowing token transfer for deposit
	See https://goerli.etherscan.io/tx/0x4a67d9456873a91ce7e40955387913035ca7b63593282c937786e11708dbf8d4

Sending transaction: Depositing to service registry
	See https://goerli.etherscan.io/tx/0xdbdb25346643a903b35b64891b6be0af2d41513211d4b05a8313eea3c7e2a64f

Successfully deposited to service registry

	Deposit contract address: 0x71ECE18dC357B93528D01bC9677c9DCc79fb09D4
		See https://goerli.etherscan.io/address/0x71ECE18dC357B93528D01bC9677c9DCc79fb09D4
	Deposit amount: 191
	Registration valid until: 2026-08-25T14:11

Sending transaction: Registering new URL
	See https://goerli.etherscan.io/tx/0xe62a6433fbe803164717bece833b3a19bd922a9c3d3ac610ec285309c8a6bbd3

Thank you for registering your services!
```

karlb commented on src/raiden_libs/service_registry.py:
```diff
@@ -54,6 +111,21 @@ def cli() -> None:
+    type=bool,
+    default=False,
+    help="Bypass all questions (Not recommended)",
+    is_flag=True,
```
The flag you are looking for is `hidden`.
```suggestion
@click.option(
    "--accept-all",
    type=bool,
    default=False,
    help="Bypass all questions (Not recommended)",
    is_flag=True,
    hidden=True,
```

palango replied:
Thanks, I didn't find that in the help with a quick look.

karlb commented on src/raiden_libs/service_registry.py:
```diff
@@ -98,82 +222,61 @@ def register(
     # Register if not yet done
     if not currently_registered:
-        deposit_to_registry(
+        # if not currently_registered:
```
This is already in the line above.
```suggestion
```

palango commented:
My proposal would be to merge this like this and create a follow-up issue that list further improvements.

The remaining issue is testing. As I changed the structure, I also removed the tests. And as it is, this code is hard to test. Any ideas?
